ChangeLog.d/fix_some_resource_leaks.txt

```diff
@@ -0,0 +1,4 @@
+Bugfix
+   * Fixed resource leaks in mbedtls_pk_parse_public_key(), 
+     mbedtls_ssl_cache_set() for certain error conditions,
+     flagged by a static analyzer.
```

Comment:
“flagged by a static analyzer” doesn't convey any useful information here. We know that the issues can happen in practice, so “fixed resource leaks … for certain error conditions” is sufficient.

Also please ensure that `tests/scripts/check_files.py` passes.